### PR TITLE
ClangImporter: run Clang with a proper executable path (pt 2)

### PR DESCRIPTION
**Pull request by egorzhdan**
This change re-applies https://github.com/apple/swift/pull/36749 after it has been reverted in https://github.com/apple/swift/pull/37805 because of a broken standalone stdlib build.

This time let's not try to pass a `-tools-directory` argument to the Swift compiler in CMake if `SWIFT_NATIVE_CLANG_TOOLS_PATH` is not defined.

### CODE REVIEW

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please test Windows platform

**Comment by egorzhdan** [comment]
@swift-ci please smoke test Windows platform

**Comment by compnerd** [comment]
I think that honoring `-tools-directory` is definitely an improvement.  I'd like to actually test this before explicitly signing off on it, but I do agree with the spirit of the change.

**Comment by egorzhdan** [comment]
@compnerd yeah that's fair. I added a test to make sure that ClangImporter actually uses the path provided as `-tools-directory`.

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
It seems the test that I've added doesn't really work for Linux, since the stdlib headers are located under a directory with a different name, which is setup-dependent (`include/c++/7` where `7` is the GCC version, instead of `include/c++/v1`).
I restricted the test to macOS for now.
I'll try to come up with a way to test this on Linux as well.

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
@egorzhdan can you try `-Xclang -stdlib=libc++` as a means of avoiding disabling the test on Linux?  In the mean time, you should probably keep the test enabled on Windows, so you should instead `XFAIL` it on Linux, Android, and BSD.

**Comment by egorzhdan** [comment]
Thanks for the suggestion @compnerd! This makes the test work properly on my local Ubuntu machine.
(I ended up using `-Xcc` instead of `-Xclang` since `swift-ide-test` doesn't know what `-Xclang` is)

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please smoke test Windows platform

**Comment by egorzhdan** [comment]
@swift-ci please smoke test Windows platform

**Comment by shahmishal** [comment]
@swift-ci please test Windows

**Comment by egorzhdan** [comment]
The Clang driver [doesn't support](https://github.com/apple/llvm-project/blob/aa674a9624777753c508dbc11d568ebcee86d1b3/clang/lib/Driver/ToolChains/MSVC.cpp#L1326) `-stdlib=libc++` on Windows at the moment, so I added `XFAIL: OS=windows-msvc`.
Fortunately there seems to be [ongoing work](https://reviews.llvm.org/D101479) to add support for it on Windows.

@compnerd do you think this would be good enough for now?

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please smoke test Linux

**Comment by egorzhdan** [comment]
@swift-ci please smoke test macOS

**Comment by egorzhdan** [comment]
Rebased this PR to fix a test failure caused by a diverged main branch.

@swift-ci please smoke test

**Review by zoecarver — APPROVED** [review]
Please wait for @compnerd's approval before landing. 

@compnerd can you please test this to verify we can ship it. I'd really like to land this patch so that we can add a test for importing the stdlib that runs on the CI.

**Comment by zoecarver** [comment]
>  I'd really like to land this patch so that we can add a test for importing the stdlib that runs on the CI.

Among other things, this will ensure we don't have any regressions in terms of what we can import.

**Comment by zoecarver** [comment]
@compnerd this patch is blocking us from being able to add some tests that use the C++ Standard Library. I'd really like to land it. 

If it's important that we run the windows standalone builds (and if there's no way to do that with swift-ci as far as I can tell) can you please run those tests locally in the next week-ish?

**Comment by egorzhdan** [comment]
Thanks @compnerd! Looks like the CI failed because of some network problem:
```
This job was abandoned. We have detected that logs from the agent may have not finished uploading. We have included our in-memory record of all log lines uploaded before we lost contact with the agent:
```
Could you please try running it again?

**Comment by zoecarver** [comment]
Thanks, Saleem 🙏

**Comment by compnerd** [comment]
Retriggered the job on the same run.

**Comment by egorzhdan** [comment]
Thanks @compnerd! The toolchain & SDK builds have passed, but the devtools build has failed for swift-argument-parser:  looks like your recent fix hasn't made its way into the build (https://github.com/apple/swift-argument-parser/commit/eef1f7590aa2578305daae01daf7cf4e482e24c8).
Since devtools & the rest of the build aren't affected by this change, this PR should probably be good to be merged now. If you don't think so, could you please rerun the devtools build with the recent swift-argument-parser?

**Comment by zoecarver** [comment]
@compnerd friendly ping. Is this good to go?

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by zoecarver** [comment]
Let's go ahead and land this. It sounds like any failures are unrelated. And the Swift CI is passing.

Thanks, Egor.

**Review by zoecarver — APPROVED** [review]

**Comment by compnerd** [comment]
@zoecarver - the Swift CI doesn't test Windows sufficiently.  That said, if the toolchain and SDK builds completed, I think that this should be fine.

**Pull request merged**
